Updated circuit generation process (#1433)
o1-labs/mina-rust: website/docs/developers/circuits.md

- Protocol compliance validation
- Regression testing for circuit changes

### Circuit Generation Process

> Since these constraint capabilities are missing, mina rust nodes requires

Reviewer: ```suggestion
Since these constraint capabilities are missing, Mina rust nodes requires
```

> Since these constraint capabilities are missing, mina rust nodes requires

Reviewer: ```suggestion
Since these constraint capabilities are missing, the Mina rust nodes require
```

externally generated circuit data. The following process describes how circuits
are created and distributed using the original Mina codebase:

> <!-- TODO:

Reviewer: As we do on the release process page, you can add a block "alert warning" to have something like this: 

<img width="1018" height="366" alt="Image" src="https://github.com/user-attachments/assets/33e362a7-8f72-4d43-89a5-f65d00067816" />.

Could you update please? It informs the reader that it is still in progress. Comments are not visible to readers.

    Update this when new mina release happens that contains code to export circuits
    And when command to export circuits is added
    And CI is updated to check for latest circuits
 -->

1. Build mina <b>OCAML</b> node from source with commit after
   [`6961849`](https://github.com/MinaProtocol/mina/commit/6961849f17d564c39e7d45e01e3ddda9a09602a4)

2. Running the circuit generation process using the branch above
   - Launch the OCaml node which produces circuit cache data in
     `/tmp/coda_cache_dir`

>    - The branch dumps the usual circuit data plus extra data specifically

Reviewer: Can we be more explicit what it generates exactly?

>      required by mina rust nodes

Reviewer: ```suggestion
     required by the Mina rust nodes
```

>    - The process also dumps blocks for use in tests

Reviewer: Can we add more context for this? Which tests? For which purposes?
